Pi2 g1 57

AnaLauraS opened the pull request:
Todas las APIs de facturacion funcionando.
Solo falta agregar el feing con el MS - Users

VERIFICAR SCRIPT DE BBDD!!

karenCespedesR committed:
api Shipping con getBy y delete

karenCespedesR committed:
api envio andando

karenCespedesR committed:
Merge branch 'Develop' of https://github.com/Daniel-Zemanate/PI2G1-back-costume-mania into PI2G1-56

karenCespedesR committed:
api sale actualizando entidad sale

karenCespedesR committed:
api Sale avance en Feign

AnaLauraS committed:
Componentes de factura funcionando

AnaLauraS committed:
GETS de factura: todas las ventas, por status, por modelo

AnaLauraS committed:
GETS de factura con feign a catalogo: by model, by catalog

AnaLauraS committed:
GETS de factura con feign a catalogo: by size. Todos los gets terminados

AnaLauraS committed:
pruebas para hacer facturas

AnaLauraS committed:
pruebas para hacer facturas OK + modificacion de campo en BBDD

AnaLauraS committed:
Creacion de factura inicial

AnaLauraS committed:
Creacion de factura inicial: resta stock

AnaLauraS committed:
Creacion de factura OK

AnaLauraS committed:
Creacion de factura OK

AnaLauraS committed:
get por facturas

AnaLauraS committed:
API para cancelar factura

AnaLauraS committed:
API para modificar factura (administrador)

AnaLauraS committed:
get by dates

Daniel-Zemanate commented:
Se debe tener en cuenta:
- requiredEntity: Hay que validar las anotaciones que se usan, veo que implementa feign, no deberían tener anotaciones de @Entity o @Table esas entidades deberían ser un modelo de referencia más parecido a un dto, usar @JsonIgnoreProperties ayudaría a ignorar propiedades que no sean solicitadas.

- User: Toda la configuración de la entidad ya está en Develop para seleccionar los atributos necesarios para el Feign, también si se necesita algún endpoint adicional se puede comentar para añadirlo.

- Archivos de configuración: actualizar el ms-bills.yml de la configuración y el application.yml del microservicio ms-bills (Se puede usar el ms-catalog de referencia), también copiar/adicionar el Dockerfile, agregar en el docker-compose que se levante ese microservicio, finalmente agregar cualquier variable de entorno en el archivo .env-aws, todo esto debería ser copiar y pegar de otro ms de referencia como ms-catalog.

- El sql script está muy desactualizado a la versión que está funcionando en Develop y cuando se ejecuta el script en local sale con errores. Recomiendo copiar el contenido del archivo en Develop, comparar y agregar los detalles necesarios para los cambios aplicados de lo contrario no va a cargar el pipeline o se cae la info de la EC2

AnaLauraS committed:
requiredEntity -> no es posible la corrección.
User -> pendiente a aplicar en próxima rama
Archivos de configuración -> hecho
El sql script -> actualizado

AnaLauraS commented:
Dani te contesto cada punto:

- required entity: no es posible borrar esas notaciones porque sino lo levanta el MS. No puede vincularse correctamente a la base datos. Error que sale: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class path resource [org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaConfiguration.class]: Association 'com.costumemania.msbills.model.Sale.catalog' targets an unknown entity named 'com.costumemania.msbills.model.requiredEntity.Catalog'
- MS-User: Una vez que se mergee esta rama a develop, configuro lo que falta en una rama nueva. Asi ya me aseguro de tener todo lo último del MS-user.
- Archivos de configuración: POR FAVOR REVISAR. Hice los cambios requeridos, pero no puedo correr Docker como para probarlo. Ademas agregué la configuración en el gateway.
- Desconozco por qué aun trae problemas ese script: lo revisé a detalle y solo tenía diferencias con las tabla de favoritos. Ya está corregido. Pero realmente no debería traerte problemas. Avisame y lo vemos juntos a ver si podemos detectar que tabla o columna genera el conflicto.

AnaLauraS committed:
nombre de columna cambiado

AnaLauraS committed:
script de BBDD corregido

Daniel-Zemanate committed:
Merge branch 'Develop' into PI2G1-57

Daniel-Zemanate commented on ms-bills/src/main/java/com/costumemania/msbills/model/requiredEntity/User.java:
Este archivo va a generar conflictos al subirlo porque no coincide con lo que hay en la base de datos. Entrar en ms-users y revisar la entidad

AnaLauraS replied:
Directamente lo elimino Dani porque la rama no hace uso de esa entidad aun.

AnaLauraS replied:
Listo Dani para el mergeo. No debería traer problemas porque la clase está vacía

AnaLauraS committed:
Update User.java
